Validation/Geometry/src/MaterialBudgetHGCalHistos.cc

```diff
@@ -0,0 +1,855 @@
+	hmgr->getHistoProf2(10472)->Fill(z,polarRadius, iMne * fillValue * cos( 2 * atan(exp(-fabs(eta))) ) );
+
+	// Silicon
+	hmgr->getHistoProf1(10540)->Fill(polarRadius,theTotalIL_NIM);
+	hmgr->getHistoProf2(10550)->Fill(z,polarRadius,theTotalIL_NIM);
+	hmgr->getHistoProf2(10552)->Fill(z,polarRadius,theTotalIL_NIM * cos( 2 * atan(exp(-fabs(eta))) ) );
+	hmgr->getHisto2(10560)->Fill(z,polarRadius,iSil*fillValue);
+	hmgr->getHistoProf2(10570)->Fill(z,polarRadius,iSil*fillValue);
+	hmgr->getHistoProf2(10572)->Fill(z,polarRadius, iSil * fillValue * cos( 2 * atan(exp(-fabs(eta))) ) );
+
+	// Other
+	hmgr->getHistoProf1(10640)->Fill(polarRadius,theTotalIL_OTH);
+	hmgr->getHistoProf2(10650)->Fill(z,polarRadius,theTotalIL_OTH);
+	hmgr->getHistoProf2(10652)->Fill(z,polarRadius,theTotalIL_OTH * cos( 2 * atan(exp(-fabs(eta))) ) );
+	hmgr->getHisto2(10660)->Fill(z,polarRadius,iOth*fillValue);
+	hmgr->getHistoProf2(10670)->Fill(z,polarRadius,iOth*fillValue);
+	hmgr->getHistoProf2(10672)->Fill(z,polarRadius, iOth * fillValue * cos( 2 * atan(exp(-fabs(eta))) ) );
+
+	// Air
+	hmgr->getHistoProf1(10740)->Fill(polarRadius,theTotalIL_AIR);
+	hmgr->getHistoProf2(10750)->Fill(z,polarRadius,theTotalIL_AIR);
+	hmgr->getHistoProf2(10752)->Fill(z,polarRadius,theTotalIL_AIR * cos( 2 * atan(exp(-fabs(eta))) ) );
+	hmgr->getHisto2(10760)->Fill(z,polarRadius,iAir*fillValue);
+	hmgr->getHistoProf2(10770)->Fill(z,polarRadius,iAir*fillValue);
+	hmgr->getHistoProf2(10772)->Fill(z,polarRadius, iAir * fillValue * cos( 2 * atan(exp(-fabs(eta))) ) );
+
+	// StainlessSteel
+	hmgr->getHistoProf1(10840)->Fill(polarRadius,theTotalIL_SST);
+	hmgr->getHistoProf2(10850)->Fill(z,polarRadius,theTotalIL_SST);
+	hmgr->getHistoProf2(10852)->Fill(z,polarRadius,theTotalIL_SST * cos( 2 * atan(exp(-fabs(eta))) ) );
+	hmgr->getHisto2(10860)->Fill(z,polarRadius,iSst*fillValue);
+	hmgr->getHistoProf2(10870)->Fill(z,polarRadius,iSst*fillValue);
+	hmgr->getHistoProf2(10872)->Fill(z,polarRadius, iSst * fillValue * cos( 2 * atan(exp(-fabs(eta))) ) );
+
+	// WCu
+	hmgr->getHistoProf1(10940)->Fill(polarRadius,theTotalIL_WCU);
+	hmgr->getHistoProf2(10950)->Fill(z,polarRadius,theTotalIL_WCU);
+	hmgr->getHistoProf2(10952)->Fill(z,polarRadius,theTotalIL_WCU * cos( 2 * atan(exp(-fabs(eta))) ) );
+	hmgr->getHisto2(10960)->Fill(z,polarRadius,iWcu*fillValue);
+	hmgr->getHistoProf2(10970)->Fill(z,polarRadius,iWcu*fillValue);
+	hmgr->getHistoProf2(10972)->Fill(z,polarRadius, iWcu * fillValue * cos( 2 * atan(exp(-fabs(eta))) ) );
+
+	// Lead
+	hmgr->getHistoProf1(11040)->Fill(polarRadius,theTotalIL_LEA);
+	hmgr->getHistoProf2(11050)->Fill(z,polarRadius,theTotalIL_LEA);
+	hmgr->getHistoProf2(11052)->Fill(z,polarRadius,theTotalIL_LEA * cos( 2 * atan(exp(-fabs(eta))) ) );
+	hmgr->getHisto2(11060)->Fill(z,polarRadius,iLea*fillValue);
+	hmgr->getHistoProf2(11070)->Fill(z,polarRadius,iLea*fillValue);
+	hmgr->getHistoProf2(11072)->Fill(z,polarRadius, iLea * fillValue * cos( 2 * atan(exp(-fabs(eta))) ) );
```

Comment:
please, calculate the expression cos( 2 * atan(exp(-fabs(eta))) ) once in advance instead of calculating it 22 times